components/cli/pkg/commands/hub/login.go

```diff
@@ -59,115 +57,84 @@ func RunLogin(registryURL string, username string, password string) {
+					return fmt.Errorf("error occurred while reading Credentials, %v", err)
 				}
-
-				spinner = util.StartNewSpinner("Logging into Cellery Registry " + registryURL)
-				err = validateCredentialsWithRegistry(registryCredentials)
-				if err != nil {
-					runPreExitWithErrorTasks(spinner)
-					if strings.Contains(err.Error(), "401") {
-						util.ExitWithErrorMessage("Invalid Credentials", err)
-					} else {
-						util.ExitWithErrorMessage("Error occurred while initializing connection to the Cellery Registry",
-							err)
-					}
+				if err = cli.ExecuteTask(fmt.Sprintf("Logging into Cellery Registry %s", registryURL),
+					fmt.Sprintf("Failed logging into Cellery Registry %s", registryURL), "", func() error {
+						err = validateCredentialsWithRegistry(registryCredentials)
+						if err != nil {
+							cli.CredReader().Shutdown(false)
+							if strings.Contains(err.Error(), "401") {
+								return fmt.Errorf("invalid Credentials, %v", err)
+							} else {
+								return fmt.Errorf("error occurred while initializing connection to the Cellery Registry, %v",
+									err)
+							}
+						}
+						isCredentialsAlreadyPresent = false // To save the new credentials
+						return nil
+					}); err != nil {
+					return err
 				}
-				isCredentialsAlreadyPresent = false
 			} else {
-				util.ExitWithErrorMessage("Invalid Credentials", err)
+				return fmt.Errorf("invalid Credentials, %v", err)
 			}
 		} else {
-			util.ExitWithErrorMessage("Error occurred while initializing connection to the Cellery Registry",
+			return fmt.Errorf("error occurred while initializing connection to the Cellery Registry, %v",
 				err)
 		}
 	}
 
 	if !isCredentialsAlreadyPresent {
 		// Saving the credentials
-		spinner.SetNewAction("Saving credentials")
-		err = credManager.StoreCredentials(registryCredentials)
-		if err != nil {
-			runPreExitWithErrorTasks(spinner)
-			util.ExitWithErrorMessage("Error occurred while saving Credentials", err)
+		if err = cli.ExecuteTask("Saving credentials", "Failed to save credentials", "", func() error {
+			err = cli.CredManager().StoreCredentials(registryCredentials)
+			if err != nil {
+				cli.CredReader().Shutdown(false)
+				return fmt.Errorf("error occurred while saving Credentials, %v", err)
+			}
+			return nil
+		}); err != nil {
+			return err
 		}
 	}
-	if isAuthorized != nil {
-		isAuthorized <- true
-	}
-	if done != nil {
-		<-done
-	}
-	spinner.Stop(true)
+	cli.CredReader().Shutdown(true)
 	util.PrintSuccessMessage(fmt.Sprintf("Successfully logged into Registry: %s", util.Bold(registryURL)))
+	return nil
 }
 
 // requestCredentialsFromUser requests the user for credentials and returns the channels (or nil) created
-func requestCredentialsFromUser(registryCredentials *credentials.RegistryCredentials) (chan bool, chan bool, error) {
-	var isAuthorized chan bool
-	var done chan bool
+func requestCredentialsFromUser(cli cli.Cli, registryCredentials *credentials.RegistryCredentials) error {
```

Comment:
Do we need a separate method now ?

Reply:
Updated and removed the unnecessary method.